sklearn/impute/_iterative.py

```diff
@@ -272,15 +272,17 @@ def _impute_one_feature(self,
             estimator = clone(self._estimator)
 
         if fit_mode:
+            indices_row_available = np.flatnonzero(~missing_row_mask)
```

Comment:
using integer indices instead of mask here

Reply:
why?

Reply:
We should not use a boolean mask to select rows in `safe_indexing`. We were not supposed to support it (even if it was working apparently). Now, it raises a proper error.

Therefore the code is adapted to go from mask to indices to select samples.